crates/flux-fhir-analysis/src/conv.rs

```diff
@@ -225,7 +225,8 @@ pub(crate) fn adt_def_for_enum(
     let def_id = enum_def.owner_id;
     let sort_args = sort_args_for_adt(genv, def_id);
     let sort = rty::Sort::tuple(conv_sorts(genv, &genv.index_sorts_of(def_id, &sort_args)));
-    let adt_def = lowering::lower_adt_def(&genv.tcx.adt_def(enum_def.owner_id));
+    let is_extern = enum_def.extern_id.is_some();
+    let adt_def = lowering::lower_adt_def(&genv.tcx.adt_def(enum_def.owner_id), is_extern);
```

Comment:
I was thinking something more like:

```suggestion
    let adt_def = if let Some(extern_id) = enum_def.extern_id {
        lowering::lower_adt_def(&genv.tcx.adt_def(extern_id))
    } else {
        lowering::lower_adt_def(&genv.tcx.adt_def(enum_def.owner_id))
    };
```

We should create the `AdtDef` with the (real) extern id if there's one.

Reply:
oh, that's much better, thanks!